rfcs/text/i18n.md

```diff
@@ -0,0 +1,270 @@
+
+# Basic example
+
+The proposal is to introduce a new component `I18nProvider` that allows our consumers to provide a 
+[`react-i18next`](https://github.com/i18next/react-i18next) compatible<sup>[1] [2]</sup> `t` function.
+
+We will need to check each instance that we use our current `i18n-js` implementation and ensure that the
+* key makes sense
+* translations are provided in each of our supported languages
+* interpolation/plural/formatting works correctly
+
+`react-i18next` also uses [`Suspense`](https://reactjs.org/docs/concurrent-mode-suspense.html) to allow for async translation loading. We don't need to support this within `carbon-react` to adopt this approach, but we will use this in the future when we add `loading` states to our components.
+
+[1]: https://react.i18next.com/latest/usetranslation-hook
+[2]: https://www.i18next.com/overview/api#t
+
+# Motivation
+
+We currently only ship english translations. Each time a product launches in a non-english market they have to set up
+translations, these are typically non-contextual and will be the same in each product.
+
+Having `i18n-js` as a peer-dependency means that we're encouraging the use of this, unintentionally, for content translations.
+The current recommended setup is `i18next`, so consumers will have to configure both libraries, increasing the bundle size and reducing performance.
+
+# Detailed design
+
+```js
+// carbon-react/lib/components/__internal__/I18nContext/index.js
+import {createContext} from "react";
+
+export default createContext(() => {throw new Error("No I18nProvider exists.")});
+```
+
+```js
+// carbon-react/lib/components/I18nProvider/index.js
+import React from "react";
+import Context from "../__internal__/I18nContext";
+export default ({t, children}) => <Context.Provider value = {t}>{children}</Context.Provider>
+```
+
+```js
+// carbon-react/lib/components/Example/index.js
+import React, { Suspense, useContext } from "react";
+import Context from "../__internal__/I18nContext";
+export default () => {
+  const t = useContext(Context);
+  return <span>{t("example.next")}</span>;
+};
+```
+
+Below is an example of how someone could configure lazy loading of carbon translations.
+This file is accessible from `/locales/fr-FR/carbon.json`
+
+```json
+{
+  "example": {
+    "next": "suivante",
+    "previous": "précédente"
+  }
+}
+```
+
+```js
+// App.js
+import React, { Suspense } from "react";
+import HttpApi from "i18next-http-backend";
+import i18n from "i18next";
+import { initReactI18next, useTranslation } from "react-i18next";
+import Example from "carbon-react/lib/components/Example";
+import I18nProvider from "carbon-react/lib/components/I18nProvider";
+
+i18n
+  .use(initReactI18next)
+  .use(HttpApi)
+  .init({
+    interpolation: {
+      escapeValue: false,
+    },
+    lng: "fr-FR",
+    ns: ["carbon", "app"],
+    backend: {
+      loadPath: "/locales/{{lng}}/{{ns}}.json",
```

Comment:
More of a question but I've never seen this method of string interpolation in js, how does this work?